File: MMORPG_2/Assets/Script/Data/LocalData/CreateLua/NPCDataEntity.lua
NPCDataEntity = { Id = 0, Name = "", PrefabName = "", RoleBirthPos = 0, HeadPicAtlas = "", HeadPic = "", HalfBodyPath = "", Talk = "" }

--这句是重定义元表的索引，就是说有了这句，这个才是一个类
NPCDataEntity.__index = NPCDataEntity;

function NPCDataEntity.New(Id, Name, PrefabName, RoleBirthPos, HeadPicAtlas, HeadPic, HalfBodyPath, Talk)
    local self = { }; --初始化self
    setmetatable(self, NPCDataEntity); --将self的元表设定为Class

    self.Id = Id;
    self.Name = Name;
    self.PrefabName = PrefabName;
    self.RoleBirthPos = RoleBirthPos;
    self.HeadPicAtlas = HeadPicAtlas;
    self.HeadPic = HeadPic;
    self.HalfBodyPath = HalfBodyPath;
    self.Talk = Talk;

    return self;
end

File: GameServer/GameServerApp/GameServerApp/Data/LocalData/Create/NPCDataEntity.cs

//===================================================
//作    者：
//创建时间：2019-01-10 09:41:31
//备    注：此代码为工具生成 请勿手工修改
//===================================================
using System.Collections;

/// <summary>
/// NPCData实体
/// </summary>
public partial class NPCDataEntity : AbstractEntity
{
    /// <summary>
    /// 名称
    /// </summary>
    public string Name { get; set; }

    /// <summary>
    /// 场景名称
    /// </summary>
    public string PrefabName { get; set; }

    /// <summary>
    /// NPC坐标
    /// </summary>
    public float[] RoleBirthPos { get; set; }

    /// <summary>
    /// 头像所在图集
    /// </summary>
    public string HeadPicAtlas { get; set; }

    /// <summary>
    /// 头像
    /// </summary>
    public string HeadPic { get; set; }

    /// <summary>
    /// 半身像所在路径
    /// </summary>
    public string HalfBodyPath { get; set; }

    /// <summary>
    /// 自言自语
    /// </summary>
    public string Talk { get; set; }

}
